src/WebJobs.Script/Description/BindingMetadata.cs

```diff
@@ -3,14 +3,37 @@
 
 namespace Microsoft.Azure.WebJobs.Script.Description
 {
+    /// <summary>
+    /// Represents the metadata for a function parameter binding.
+    /// </summary>
     public class BindingMetadata
     {
+        /// <summary>
+        /// Gets or sets the name of the binding.
+        /// </summary>
         public string Name { get; set; }
 
+        /// <summary>
+        /// Gets or sets an app setting name for a setting whose value
+        /// contains the connection details that should be used for this binding.
+        /// E.g. for a Queue binding, this might be set to the name of a an app
+        /// setting containing the Azure Storage connection string to use.
+        /// </summary>
+        public string Connection { get; set; }
```

Comment:
The new metadata property. The idea is that the moniker "connection" is general enough to be usable for non Azure Storage scenarios as well.

For example, the core SDK also has a corresponding ServiceBusAccountAttribute that works the same way as StorageAccountAttribute. I haven't added support for that in this PR, but we eventually will. When we do, ServiceBus bindings will use this same property to specify an app setting name that points to their SB connection string.